src/erleans_stream_manager.erl

```diff
@@ -156,31 +147,15 @@ handle_info({StreamRef, {go, _Ref, Pid, _RelativeTime, _SojournTime}}, State=#st
 handle_info({_Stream, {drop, _SojournTime}}, State) ->
     %% should never happen... we have an infinite timeout
     {noreply, State};
-handle_info(update_streams, State=#state{ring=Ring,
-                                         provider={ProviderModule, Provider}}) ->
+handle_info(update_streams, State=#state{provider=Provider}) ->
     lager:info("at=update_streams", []),
-    {ok, Streams} = ProviderModule:all(erleans_stream, Provider),
-
-    %% TODO: oh, so inefficient
     %sbroker:dirty_cancel(?STREAM_BROKER, ?STREAM_TAG),
-    MyStreams = lists:foldl(fun({StreamRef, erleans_stream, ETag, Subscribers}, Acc) ->
-                                enqueue_if_node(StreamRef, Subscribers, ETag, Ring, Acc)
-                            end, #{}, Streams),
-    {noreply, State#state{streams = MyStreams,
-                          tref = undefined}};
-handle_info({update, Membership}, State=#state{tref=TRef}) ->
-    cancel_timer(TRef),
-    TRef1 = erlang:send_after(?TIMEOUT, self(), update_streams, []),
-    %% convert partisan membership to list of erlang nodes
-    Members = [P || {P, _, _} <- sets:to_list(state_orset:query(Membership))],
-    NewRing = make_ring(Members),
-    {noreply, State#state{ring = NewRing,
-                          tref = TRef1}};
+    Range = erleans_partitions:get_range(),
+    MyStreams = enqueue_streams(Provider, Range),
+    {noreply, State#state{streams = MyStreams}};
 handle_info(timeout, State) ->
-    TRef = erlang:send_after(?TIMEOUT, self(), update_streams, []),
-    Ring = make_ring(),
-    {noreply, State#state{ring = Ring,
-                          tref = TRef}}.
+    self() ! update_streams,
```

Comment:
if you changed the message from the partitions server to contain the range, you'd only need to fetch it here.